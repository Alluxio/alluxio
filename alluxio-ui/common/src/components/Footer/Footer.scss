.footer {
  font-size: 0.9em;

  .nav-link {
    padding: 0.33em 1em;
  }
}
