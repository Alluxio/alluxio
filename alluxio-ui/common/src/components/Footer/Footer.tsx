/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import React from 'react';
import {Nav, NavItem, NavLink} from 'reactstrap';

import {INavigationData, INavigationDataCallbackParameters} from '../../constants';

import './Footer.css';

export interface IFooterProps {
  data: INavigationData[];
  callbackParameters?: INavigationDataCallbackParameters;
}

export class Footer extends React.PureComponent<IFooterProps> {
  constructor(props: IFooterProps) {
    super(props);

    this.renderNavItems = this.renderNavItems.bind(this);
  }

  public render(): JSX.Element {
    const {data} = this.props;
    return (
      <div className="footer mt-auto navbar-dark bg-primary">
        <Nav className="justify-content-center">
          {this.renderNavItems(data)}
        </Nav>
      </div>
    );
  }

  private renderNavItems(datas: INavigationData[]) {
    const {callbackParameters} = this.props;
    return datas.map((data: INavigationData) => {
      const url = typeof data.url === 'function'
        ? (callbackParameters ? data.url(callbackParameters) : data.url({}))
        : data.url;
      return (
        <NavItem key={url}>
          <NavLink href={url}>{data.innerText}</NavLink>
        </NavItem>
      )
    });
  }
}
