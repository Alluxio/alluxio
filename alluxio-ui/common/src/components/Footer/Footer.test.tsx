/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {configure, mount, ReactWrapper, shallow, ShallowWrapper} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import {StaticRouter} from 'react-router';

import {Footer, IFooterProps} from './Footer';

configure({adapter: new Adapter()});

describe('Footer', () => {
  let props: IFooterProps;

  beforeAll(() => {
    props = {
      data: []
    };
  });

  describe('Shallow component', () => {
    let shallowWrapper: ShallowWrapper;

    beforeAll(() => {
      shallowWrapper = shallow(<Footer {...props}/>);
    });

    it('Renders without crashing', () => {
      expect(shallowWrapper.length).toEqual(1);
    });

    it('Matches snapshot', () => {
      expect(shallowWrapper).toMatchSnapshot();
    });
  });

  describe('React component', () => {
    let reactWrapper: ReactWrapper;
    let context = {};

    beforeAll(() => {
      reactWrapper = mount(
        <StaticRouter location="someLocation" context={context}>
          <Footer {...props}/>
        </StaticRouter>
      );
    });

    it('Renders without crashing', () => {
      expect(reactWrapper.length).toEqual(1);
    });

    it('Matches snapshot', () => {
      expect(reactWrapper).toMatchSnapshot();
    });
  });
});
