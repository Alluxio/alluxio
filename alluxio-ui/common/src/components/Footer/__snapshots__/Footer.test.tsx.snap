// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer React component Matches snapshot 1`] = `
<StaticRouter
  basename=""
  context={Object {}}
  location="someLocation"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "someLocation",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Footer
      data={Array []}
    >
      <div
        className="footer mt-auto bg-secondary"
      >
        <Nav
          className="justify-content-center"
          tag="ul"
          vertical={false}
        >
          <ul
            className="justify-content-center nav"
          />
        </Nav>
      </div>
    </Footer>
  </Router>
</StaticRouter>
`;

exports[`Footer Shallow component Matches snapshot 1`] = `
<div
  className="footer mt-auto bg-secondary"
>
  <Nav
    className="justify-content-center"
    tag="ul"
    vertical={false}
  />
</div>
`;
